en/tutorials-and-examples/cms/authentication.rst

```diff
@@ -1,8 +1,142 @@
 CMS Tutorial - Authentication
 #############################
 
-* Adding login
-* Adding logout
+Now that our CMS has users, we should enable them to login, and apply some basic
+access control to the article creation & editing experiences.
+
+Adding Login
+============
+
+In CakePHP, authentication is handled by :doc:`/controllers/components`.
+Components can be thought of as ways to create reusable chunks of controller
+code related to a specific feature or concept. Components can hook into the
+controller's event life-cycle and interact with your application that way. To
+get started, we'll add the :doc:`AuthComponent
+</controllers/components/authentication>` to our application. We'll want the
+create, update and delete methods to require authentication, so we'll add
+AuthComponent in our AppController::
+
+    // In src/Controller/AppController.php
+    namespace App\Controller;
+
+    use Cake\Controller\Controller;
+
+    class AppController extends Controller
+    {
+        public function initialize()
+        {
+            $this->loadComponent('Flash');
+            $this->loadComponent('Auth', [
+                'authenticate' => [
+                    'Form' => [
+                        'fields' => [
+                            'username' => 'email',
+                            'password' => 'password'
+                        ]
+                    ]
+                ],
+                'loginAction' => [
+                    'controller' => 'Users',
+                    'action' => 'login'
+                ],
+                 // If unauthorized, return them to page they were just on
+                'unauthorizedRedirect' => $this->referer()
+            ]);
+
+            // Allow the display action so our pages controller
+            // continues to work. Also enable the read only actions.
+            $this->Auth->allow(['display', 'view', 'index']);
+        }
+    }
```

Comment:
As this was baked, it has code it in at this point. Do you think including that code or a `// existing code` would be good? If using the `// existing code`, you can remove the FlashComponent as that is baked in. Here's the baked code if you want to include it instead:

```php
    public function initialize()
    {
        parent::initialize();

        $this->loadComponent('RequestHandler');
        $this->loadComponent('Flash');

        /*
         * Enable the following components for recommended CakePHP security settings.
         * see http://book.cakephp.org/3.0/en/controllers/components/security.html
         */
        //$this->loadComponent('Security');
        //$this->loadComponent('Csrf');
    }
```